notes.md

```diff
@@ -0,0 +1,215 @@
+```
+
+# Cannot pack columns into a named tuple column?
+If I can do this to unpack named tuples:
+```julia
+julia> df = DataFrame(in=[(a=1,b=2), (a=3, b=4)])
+2×1 DataFrame
+ Row │ in
+     │ NamedTup…
+─────┼────────────────
+   1 │ (a = 1, b = 2)
+   2 │ (a = 3, b = 4)
+
+julia> df2 = select(df, :in => AsTable)
+2×2 DataFrame
+ Row │ a      b
+     │ Int64  Int64
+─────┼──────────────
+   1 │     1      2
+   2 │     3      4
+```
+Then I should also be able to do the reverse to pack named tuples and get `df` back:
+```julia
+julia> df3 = select(df2, AsTable(:) => :in)
+ERROR: ArgumentError: Unrecognized column selector: AsTable(Colon()) => :in
+```
+
+# Why are types truncated when there is still a lot of room?
+
+Also, why is `NamedTup...` truncated in the output above? There is room.
+
+# AsTable cannot be used on it's own?
+Maybe it should return a NamedTuple?
+
+```julia
+julia> select(df, :first_name)
+3×1 DataFrame
+ Row │ first_name
+     │ String
+─────┼────────────
+   1 │ José
+   2 │ Emma
+   3 │ Nathan
+
+julia> select(df, AsTable(:first_name))
+ERROR: ArgumentError: Unrecognized column selector: AsTable(:first_name)
+
+julia> select(df, :)
+3×4 DataFrame
+ Row │ id     first_name  last_name  age
+     │ Int64  String      String     Int64
+─────┼─────────────────────────────────────
+   1 │     1  José        Garcia        61
+   2 │     2  Emma        Marino        24
+   3 │     3  Nathan      Boyer         33
+
+julia> select(df, AsTable(:))
+ERROR: ArgumentError: Unrecognized column selector: AsTable(Colon())
+```
+
+# Is summary table correct?
+Can `subset` ever create new columns?
+Can `combine` ever create more than one row?
+
+# Add renaming function method to `rename`?
+I understand why this cannot work for the manipulation functions,
+but it seems like this method would be useful for `rename` and `rename!`.
+```julia
+julia> df = DataFrame(a=1:4, b=5:8)
+4×2 DataFrame
+ Row │ a      b
+     │ Int64  Int64
+─────┼──────────────
+   1 │     1      5
+   2 │     2      6
+   3 │     3      7
+   4 │     4      8
+
+julia> add_new(s) = "new_" * s
+add_new (generic function with 1 method)
+
+julia> rename(df, :a => add_new)
+ERROR: MethodError: no method matching rename!(::DataFrame, ::Vector{Pair{Symbol, typeof(add_new)}})
+```
+
+I would like it to behave as below:
+```julia
+julia> rename(df, :a => add_new)
+4×2 DataFrame
+ Row │ new_a  b
+     │ Int64  Int64
+─────┼──────────────
+   1 │     1      5
+   2 │     2      6
+   3 │     3      7
+   4 │     4      8
+```
+
+This works, but can only be applied to the entire data frame.
+```julia
+julia> rename(add_new, df)
+4×2 DataFrame
+ Row │ new_a  new_b
+     │ Int64  Int64
+─────┼──────────────
+   1 │     1      5
+   2 │     2      6
+   3 │     3      7
+   4 │     4      8
+```
+
+# Cannot automatically group from NamedTuples?
+I am suprised that this does not work:
+```julia
+julia> df = DataFrame(data = [(a=1,b=2), (b=3,a=4)]) # vector of `NamedTuple`s
```

Comment:
this is intentional. Why would you expect this to work? This is clearly bad data. It should be a `Dict` not a `NamedTuple` if you want the order to be ignored:
```
julia> transform(df, :data => ByRow(Dict∘pairs)=> AsTable)
2×3 DataFrame
 Row │ data            a      b
     │ NamedTup…       Int64  Int64
─────┼──────────────────────────────
   1 │ (a = 1, b = 2)      1      2
   2 │ (b = 3, a = 4)      4      3
```

Reply:
This is the only comment I'll push back on. I have the impression that order matters for tuples but not named tuples: keyword arguments can be supplied in any order, and named tuple destructuring works by name not by order. The difference between `NamedTuple` and `Dict` to me is mutability. As long as all of the named elements exist in the `data` column, I would still consider the data good. Even if not, `missing` values could be inserted as needed.
In any case, I will open a separate issue to discuss.

Reply:
Mutability is also a difference.
But note that we have `Dict` and `OrderedDict`. `OrderedDict` is kind-of mutable equivalent of `NamedTuple` (with a bit different API).

Reply:
I guess you are right.
```julia
julia> (; a=3, b=4) == (; a=3, b=4)
true

julia> (; a=3, b=4) == (; b=4, a=3)
false

julia> Dict(:a=>3, :b=>4) == Dict(:a=>3, :b=>4)
true

julia> Dict(:a=>3, :b=>4) == Dict(:b=>4, :a=>3)
true
```